### ISSUE TITLE
LTA Array.clone behaviour when original is based off a lazy iterable

### ISSUE THREAD

**Issue by zoffixznet** [rakudo/rakudo#2267]
Don't know if we can do anything about this (and if not, then this should be documented properly), but if you have an Array that:

1. was created using a lazy iterable
2. clone that array before reifying it
3. modify a cell in the original that before reifying it

Then the change happens in both the original and cloned array. However, if you reify enough **to just cover the cell you'll modify** before you clone, then the change won't get propagated to the clone.

```
<Zoffix_> m: my @a = 1,;      my @b = @a.clone; @a[40] = 100; say @b[40]
<camelia> rakudo-moar 18a6968b9: OUTPUT: «(Any)␤»
<Zoffix_> m: my @a = lazy 1,; my @b = @a.clone; @a[40] = 100; say @b[40]
<camelia> rakudo-moar 18a6968b9: OUTPUT: «100␤»
<Zoffix_> m: my @a = lazy 1,; @a[^100]; my @b = @a.clone; @a[40] = 100; say @b[40]
<camelia> rakudo-moar 18a6968b9: OUTPUT: «(Any)␤»
```

So basically any portions that were unreified during the clone remain shared until they are reified, even if the get changed by other means (I don't even know how to put this into words properly).

It seems like from user's perspective this is quite a wart and it makes array cloning rather dangerous/useless. And we can't just reify on clone, because we might be dealing with infinite arrays.
